server/auth.go

```diff
@@ -83,6 +84,14 @@ func (u *User) clone() *User {
 	clone := &User{}
 	*clone = *u
 	clone.Permissions = u.Permissions.clone()
+
+	if len(u.AllowedConnectionTypes) > 0 {
```

Comment:
dont know if this is needed, just noticed it might not be cloning this struct

Reply:
I guess this is good to do.